Add a section for how to review code more easily (#1538)
rust-lang/rustc-dev-guide: src/git.md

greatly simplifies bisecting and makes the history and commit log much easier
to follow and understand.

> ## Review suggestions

Reviewer: To me, "Review suggestions" sounds like we're telling people to review their suggestions.

```suggestion
## Tips for reviewing
```

**NOTE**: This section is for *reviewing* PRs, not authoring them.

### Hiding whitespace

> Github has a button for disabling whitespace changes we recommend using.

Reviewer: FWIW, I usually review with whitespace changes visible. Perhaps say "that can be useful" or something like that?

You can also use `git diff -w origin/master` to view changes locally.

![hide whitespace](./img/github-whitespace-changes.png)

### Fetching PRs

To checkout PRs locally, you can use `git fetch upstream pull/NNNNN/head && git checkout

> FETCH_HEAD`.

Reviewer: I've always used the `gh` CLI tool; I should try this way at some point!

You can also use github's cli tool. Github shows a button on PRs where you can copy-paste the
command to check it out locally. See <https://cli.github.com/> for more info.

![`gh` suggestion](./img/github-cli.png)

### Moving large sections of code

Git and Github's default diff view for large moves *within* a file is quite poor; it will show each
line as deleted and each line as added, forcing you to compare each line yourself. Git has an option
to show moved lines in a different color:

```

> git log -p --color-moved=dimmed-zebra --color-moved-ws=allow-indentation-change

Reviewer: Nice! I didn't think of using `log -p`.

```

> See https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---color-movedltmodegt for more info.

Reviewer: ```suggestion
See [Git's docs](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---color-movedltmodegt) for more info.
```

### range-diff

> See [the author section](#git-range-diff). This can be useful for comparing code that was

Reviewer: ```suggestion
See [the relevant section for PR authors](#git-range-diff). This can be useful for comparing code that was
```
